metis-repository/src/main/java/eu/europeana/metis/repository/rest/RecordController.java

```diff
@@ -0,0 +1,64 @@
+package eu.europeana.metis.repository.rest;
+
+import eu.europeana.metis.repository.dao.Record;
+import eu.europeana.metis.repository.dao.RecordDao;
+import eu.europeana.metis.utils.RestEndpoints;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import io.swagger.v3.oas.annotations.tags.Tags;
+import java.util.Date;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Controller for record management.
+ */
+@RestController
+@RequestMapping("records")
+@Tags(@Tag(name = RecordController.CONTROLLER_TAG_NAME,
+        description = "Controller providing access to record management functionality."))
+@Api(tags = RecordController.CONTROLLER_TAG_NAME)
+public class RecordController {
+
+  public static final String CONTROLLER_TAG_NAME = "RecordController";
+
+  private RecordDao recordDao;
+
+  @Autowired
+  void setRecordDao(RecordDao recordDao){
+    this.recordDao = recordDao;
+  }
+
+  /**
+   * It creates a record to be put into the database
+   *
+   * @param recordId - A unique record id
+   * @param datasetId - The id of the dataset which the record belongs to
+   * @param dateStamp - Last time the record was updated. It can also be the date of creation
+   * @param edmRecord - The record itself
+   * @return The record just saved in the database
+   */
+  @PostMapping(value = RestEndpoints.ADD_RECORD_TO_DATABASE, consumes = {
+      MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE}, produces = {
```

Comment:
Should it consume json? We only support XML records.

Reply:
That's a good point, I will remove the JSON consumption

Reply:
I think you actually removed XML consumption by accident?

Reply:
Oops, corrected that